### PR TITLE
[ALLUXIO-2830] Shade OSS/S3/S3a/GCS UnderFileSystem Modules

### PR DESCRIPTION
**Pull request by apc999**
https://alluxio.atlassian.net/browse/ALLUXIO-2830

### CODE REVIEW

File: underfs/oss/pom.xml [hunk 1]
```diff
@@ -40,8 +42,8 @@
       <artifactId>guava</artifactId>
     </dependency>
     <dependency>
-      <groupId>org.apache.httpcomponents</groupId>
```
Comment (jsimsa) [inline underfs/oss/pom.xml]: how is this related to your change?

File: underfs/s3a/pom.xml [hunk 2]
```diff
@@ -27,17 +28,23 @@
...
+      <artifactId>aws-java-sdk-core</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>com.amazonaws</groupId>
       <artifactId>aws-java-sdk-s3</artifactId>
     </dependency>
     <dependency>
-      <groupId>org.slf4j</groupId>
-      <artifactId>slf4j-api</artifactId>
+      <groupId>commons-codec</groupId>
```
Comment (jsimsa) [inline underfs/s3a/pom.xml]: why is this introduced?

File: underfs/oss/pom.xml (hunk 1 above)
Comment (apc999) [inline underfs/oss/pom.xml]: cleaning the deps. this `org.apache.httpcomponents` is declared but not used in oss module itself, according to `mvn dependency:analyze`

File: underfs/s3a/pom.xml (hunk 2 above)
Comment (apc999) [inline underfs/s3a/pom.xml]: it is a dep used by not declared, according to `mvn dependency:analyze`

**Comment by apc999** [comment]
@jsimsa there will be one or two more PRs. After that, I will create the script to verify the shading plugin works

**Comment by jsimsa** [comment]
sgtm
